pyccl/halos/hmfunc.py

```diff
@@ -596,15 +620,18 @@ def _setup(self, cosmo):
                 self.bz = -0.698
                 self.cz = -0.310
 
-    def _check_mdef_strict(self, mdef):
-        if isinstance(mdef.Delta, str):
+    def _check_mass_def_strict(self, mass_def):
+        if isinstance(mass_def.Delta, str):
             return True
-        elif int(mdef.Delta) == 200:
-            if (mdef.rho_type != 'matter') and \
-               (mdef.rho_type != 'critical'):
-                return True
-        elif int(mdef.Delta) == 500:
-            if mdef.rho_type != 'critical':
+        # NOTE: The following lines have been commented out because
+        # currently rho_type can only be 'matter' or 'critical' and
+        # so this case cannot be covered.
+        elif int(mass_def.Delta) == 200:
```

Comment:
but you're not checking anywhere if `rho_type` is something else, right?

Reply:
I commented it out because it wasn't covered in the unit tests. This PR is a catch-all kind of PR, so there are some instances where I've fixed bugs, coverage etc.
I could bring those lines back, but they will stay uncovered.

Reply:
I added an extra test and un-commented those lines, so this is resolved now.